easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -562,12 +568,25 @@ def _parse_dependency(self, dep, hidden=False):
             toolchain = dep.toolchain()
             if toolchain is not None:
                 dependency['toolchain'] = toolchain
+
         elif isinstance(dep, (list, tuple)):
-            # try and convert to list
-            dep = list(dep)
-            dependency.update(dict(zip(attr, dep)))
+            if dep and dep[-1] == EXTERNAL_MODULE_MARKER:
+                if len(dep) == 2:
+                    dependency['external_module'] = True
+                    dependency['short_mod_name'] = dep[0]
+                    dependency['full_mod_name'] = dep[0]
+                else:
+                    raise EasyBuildError("Incorrect external dependency specification: %s", dep)
+            else:
+                # non-external dependency: tuple (or list) that specifies name/version(/versionsuffix(/toolchain))
+                dependency.update(dict(zip(attr, dep)))
```

Comment:
it was never needed, a `zip` works fine on list/tuple mix

the `zip` doesn't make sense in the case of an external module, since we can't define `name` or `version` then, we only know the module name (the name/version can't be derived easily, or at least we shouldn't assume we can)